src/hybrid_reader.c

```diff
@@ -340,12 +340,14 @@ IndexIterator *NewHybridVectorIterator(HybridIteratorParams hParams) {
     hi->searchMode = VECSIM_STANDARD_KNN;
   } else {
     // hi->searchMode is VECSIM_HYBRID_ADHOC_BF || VECSIM_HYBRID_BATCHES
-    hi->topResults = rm_malloc(heap_sizeof(hParams.query.k));
-    heap_init(hi->topResults, cmpVecSimResByScore, NULL, hParams.query.k);
-    hi->returnedResults = array_new(RSIndexResult *, hParams.query.k);
     // Get the estimated number of results that pass the child "sub-query filter". Note that
     // this is an upper bound, and might even be larger than the total vector index size.
     size_t subset_size = hParams.childIt->NumEstimated(hParams.childIt->ctx);
+    // UINT32_MAX is the default (invalid) value for indicating invalid intersection iterator.
+    if (subset_size == UINT32_MAX) {
```

Comment:
if this is exposed (meaning, not iterator internal) have and variable/define with this value.